### PR TITLE
Create a Snuba dataset task

### PR DESCRIPTION
**Pull request by ayirr7**

### CODE REVIEW

File: snuba/migrations/snuba_migrations/audit_log/0001_auditlog.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    def forwards_local(self) -> Sequence[operations.SqlOperation]:
+        return [
+            operations.CreateTable(
+                storage_set=StorageSetKey.AUDIT_LOG,
+                table_name="audit_log_local",
+                columns=columns,
+                engine=table_engines.ReplacingMergeTree(
+                    storage_set=StorageSetKey.AUDIT_LOG,
+                    order_by="timestamp",
+                    partition_by="user",
```
Comment (lynnagara) [inline snuba/migrations/snuba_migrations/audit_log/0001_auditlog.py:29]: Any particular reason for choosing user as the partitioning key?

File: snuba/datasets/entities/audit_log.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        Column("event_id", UUID()),
+        Column("timestamp", DateTime()),
+        Column("event_type", String()),
+        Column("user", String()),
+        Column("details", String()),
+    ]
+)
+
+
+class AuditLogEntity(Entity, ABC):
```
Comment (lynnagara) [inline snuba/datasets/entities/audit_log.py:24]: I don't think this one needs to be an abstract base class, unless you planning to subclass it?

**Comment by lynnagara** [comment]
If you rebase/merge master you might notice that the `snuba_migrations` folder moved. Sorry about that - we are in the process of trying to split out the migrations framework part into a separate module that will eventually move out of Snuba.

**Comment by ayirr7** [comment]
Onboarding task completed

**Pull request closed without merging**